File: simplerecyclerviewtouchlistener/src/main/java/com/wordpress/beendora/simplerecyclerviewtouchlistener/SimpleRecyclerViewOnItemTouchListener.java
package com.wordpress.beendora.simplerecyclerviewtouchlistener;

import android.content.Context;
import android.view.GestureDetector;
import android.view.MotionEvent;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class SimpleRecyclerViewOnItemTouchListener implements RecyclerView.OnItemTouchListener {
    
    private GestureDetector mGestureDetector;

    private SimpleRecyclerViewOnItemTouchListener(Context context, Builder builder, RecyclerView recyclerView) {
        MyGestureListener gestureListener = new MyGestureListener(recyclerView,
                builder.mOnItemClickListener,
                builder.mOnItemDoubleClickListener,
                builder.mOnItemLongPressListener);
        mGestureDetector = new GestureDetector(context, gestureListener);
    }

    @Override
    public boolean onInterceptTouchEvent(@NonNull RecyclerView recyclerView,
            @NonNull MotionEvent motionEvent) {
        mGestureDetector.onTouchEvent(motionEvent);
        return false;
    }
    
    @Override
    public void onTouchEvent(@NonNull RecyclerView recyclerView, @NonNull MotionEvent motionEvent) {}
    
    @Override
    public void onRequestDisallowInterceptTouchEvent(boolean b) {}

    public static class Builder {

        private SimpleRecyclerViewOnItemTouchListener.OnItemClickListener mOnItemClickListener = null;
        private SimpleRecyclerViewOnItemTouchListener.OnItemDoubleClickListener mOnItemDoubleClickListener = null;
        private SimpleRecyclerViewOnItemTouchListener.OnItemLongPressListener mOnItemLongPressListener = null;

        public Builder setOnItemClickListener(SimpleRecyclerViewOnItemTouchListener.OnItemClickListener mOnItemClickListener) {
            this.mOnItemClickListener = mOnItemClickListener;
            return this;
        }

        public Builder setOnItemDoubleClickListener(SimpleRecyclerViewOnItemTouchListener.OnItemDoubleClickListener mOnItemDoubleClickListener) {
            this.mOnItemDoubleClickListener = mOnItemDoubleClickListener;
            return this;
        }

        public Builder setOnItemLongPressListener(SimpleRecyclerViewOnItemTouchListener.OnItemLongPressListener mOnItemLongPressListener) {
            this.mOnItemLongPressListener = mOnItemLongPressListener;
            return this;
        }

        public SimpleRecyclerViewOnItemTouchListener build(Context context, RecyclerView recyclerView) {
            return new SimpleRecyclerViewOnItemTouchListener(context, this, recyclerView);
        }

    }

    public interface OnItemClickListener {

        void onClick(View view, int position);

    }

    public interface OnItemDoubleClickListener {

        void onDoubleClick(View view, int position);

    }

    public interface OnItemLongPressListener {

        void onLongPress(View view, int position);

    }

}

File: simplerecyclerviewtouchlistener/src/main/java/io/github/horaciocome1/simplerecyclerviewtouchlistener/SimpleRecyclerViewOnItemTouchListener.kt
/*
 *    Copyright 2018 Horácio Flávio Comé Júnior
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific la
 */

package io.github.horaciocome1.simplerecyclerviewtouchlistener

import android.content.Context
import android.view.GestureDetector
import android.view.MotionEvent
import android.view.View
import androidx.recyclerview.widget.RecyclerView

class SimpleRecyclerViewOnItemTouchListener(private val context: Context) : RecyclerView.OnItemTouchListener {

    private lateinit var recyclerView: RecyclerView
    private var onClick = { _: View, _: Int -> Unit }
    private var onDoubleClick = { _: View, _: Int -> Unit }
    private var onLongPress = { _: View, _: Int -> Unit }

    override fun onTouchEvent(rv: RecyclerView, e: MotionEvent) {
        GestureDetector(context, MyGestureListener(onClick, onDoubleClick, onLongPress, recyclerView))
    }

    override fun onInterceptTouchEvent(rv: RecyclerView, e: MotionEvent): Boolean {
        return true
    }

    override fun onRequestDisallowInterceptTouchEvent(disallowIntercept: Boolean) {
    }

    fun setOnItemClickListener(onClick: (View, Int) -> Unit) {
        this.onClick = onClick
    }

    fun setOnItemDoubleClickListener(onDoubleClick: (View, Int) -> Unit) {
        this.onDoubleClick = onDoubleClick
    }

    fun setOnItemLongPressListener(onLongPress: (View, Int) -> Unit) {
        this.onLongPress = onLongPress
    }

    fun setRecyclerView(recyclerView: RecyclerView) {
        this.recyclerView = recyclerView
    }



}